src/test/java/org/cactoos/iterator/ImmutableTest.java

```diff
@@ -45,4 +46,22 @@ public void doesNotAllowRemovingOfElements() {
         immutable.next();
         immutable.remove();
     }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void decoratesNext() {
+        final Iterator<String> iter = Mockito.mock(Iterator.class);
+        final Iterator<String> immutable = new Immutable<>(iter);
+        immutable.next();
+        Mockito.verify(iter).next();
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void decoratesHasNext() {
```

Comment:
@krzyk this test doesn't test the functionality of `hasNext` method, please refactor it.